services-geocoding/src/main/java/com/mapbox/api/geocoding/v5/models/RoutablePoint.java

```diff
@@ -0,0 +1,106 @@
+package com.mapbox.api.geocoding.v5.models;
+
+import androidx.annotation.Nullable;
+
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.SerializedName;
+import com.mapbox.geojson.Point;
+
+/**
+ * Object representing {@link CarmenFeature}'s routable point.
+ *
+ * @since 6.10.0
+ */
+@AutoValue
+public abstract class RoutablePoint {
+
+  /**
+   * A string representing the routable point name.
+   *
+   * @return routable point name
+   * @since 6.10.0
+   */
+  @Nullable
+  @SerializedName("name")
+  public abstract String name();
+
+  /**
+   * A {@link Point} object which represents the routable point location.
+   *
+   * @return a GeoJson {@link Point} which defines the routable point location
+   * @since 6.10.0
+   */
+  @Nullable
+  public Point coordinate() {
+    final double[] coordinate = rawCoordinate();
+    if (coordinate != null && coordinate.length == 2) {
+      return Point.fromLngLat(coordinate[0], coordinate[1]);
+    }
+    return null;
+  }
+
+  // No public access thus, we lessen enforcement on mutability here.
+  @Nullable
```

Comment:
Same as above re: Is this really nullable?

Reply:
Discussed in another thread.